pkg/logs/logs.go

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+// Package logs provides a thin wrapper around the zerolog logging library,
+// enabling structured, context logging across an application.
+//
+// It includes:
+//   - Setting the global logging level.
+//   - Extracting loggers from context, with fallback to the global logger.
+//   - Attaching metadata (e.g., job name, correlation IDs) to loggers via context.
+//
+// This package ensures consistent logging behavior and makes it easy to trace
+// log output using structured fields like correlation IDs, even across goroutines or services.
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	AllowedLogLevels string = "disabled|trace|debug|info|warn|error|fatal|panic"
+
+	correlationIDKey = "correlation_id"
+	jobKey           = "job"
+)
+
+// SetupLogger configures the global log level for the zerolog logger.
+//
+// It takes a string `logLevel` representing the desired logging level (e.g. "debug", "info", "warn").
+// If the provided log level is invalid, it returns an error including the list of allowed levels.
+//
+// Parameters:
+//   - logLevel: a string indicating the desired log verbosity level.
+//
+// Returns:
+//   - error: if the log level cannot be parsed, an error is returned; otherwise, nil.
+//
+// Example:
+//
+//	err := SetupLogger("info")
+//	if err != nil {
+//	    log.Fatal(err)
+//	}
+func SetupLogger(logLevel string) error {
+	level, err := zerolog.ParseLevel(logLevel)
+	if err != nil {
+		return fmt.Errorf("failed parsing log level: %w; only allowed levels are: %s", err, AllowedLogLevels)
+	}
+
+	zerolog.SetGlobalLevel(level)
+	return nil
+}
+
+// Ctx retrieves the zerolog logger from the given context.
+//
+// If the context does not have a logger attached, zerolog.Ctx(ctx) returns a disabled logger.
+// In such cases, this function returns the global logger instead.
```

Comment:
A bit shortened, as the caller is not so much interested in the implementation details.
```suggestion
// If the context does not have a logger attached, the global logger is returned instead.
```

Reply:
Thanks. 
Shortened